src/core/state/selectedTransaction/index.ts

```diff
@@ -1,17 +1,26 @@
-import { createRainbowStore } from '~/core/state/internal/createRainbowStore';
+import { createBaseStore } from 'stores';
+
 import { RainbowTransaction } from '~/core/types/transactions';
 
+import { createExtensionStoreOptions } from '../_internal';
+
 export interface SelectedTransactionState {
   getSelectedTransaction: () => RainbowTransaction | null;
   setSelectedTransaction: (transaction?: RainbowTransaction) => void;
   selectedTransaction: RainbowTransaction | null;
 }
 
 export const useSelectedTransactionStore =
-  createRainbowStore<SelectedTransactionState>((set, get) => ({
-    getSelectedTransaction: () => get()?.selectedTransaction,
-    setSelectedTransaction: (selectedTransaction?: RainbowTransaction) => {
-      set({ selectedTransaction });
-    },
-    selectedTransaction: null,
-  }));
+  createBaseStore<SelectedTransactionState>(
+    (set, get) => ({
+      getSelectedTransaction: () => get()?.selectedTransaction,
+      setSelectedTransaction: (selectedTransaction?: RainbowTransaction) => {
+        set({ selectedTransaction });
+      },
+      selectedTransaction: null,
+    }),
+    createExtensionStoreOptions({
+      storageKey: 'selectedTransaction',
+      version: 0,
+    }),
```

Comment:
Was this persistence intentional? I think this is only used for UI context and designed for single-threading

  | File                    | Usage                                                     | Lifecycle |
  |-------------------------|-----------------------------------------------------------|-----------|
  | ActivityContextMenu.tsx | Sets transaction on context menu open, clears on close    | Ephemeral |
  | useActivityShortcuts.ts | Reads to enable keyboard shortcuts                        | Read-only |
  | useHomeShortcuts.ts     | Reads to disable home shortcuts when transaction selected | Read-only |
  | useCurrentHomeSheet.tsx | Reads to render SpeedUp/Cancel sheet                      | Read-only |

Reply:
nice catch!